Jamfile

```diff
@@ -56,15 +56,52 @@ rule linking ( properties * )
 {
 	local result ;
 
+	# ssl=openssl => crypto=libcrypto
+	# this match the previous behavior
+	if <ssl>openssl in $(properties)
+		&& ! <crypto>libcrypto in $(properties)
+	{
+		EXIT "with ssl=openssl the value for crypto must be libcrypto" ;
```

Comment:
the key thing with bjam is that there is no _global_ state for "the build". There is no "the build". All state (build features/configs/properties) are associated with a specific target. You can have multiple builds of the same target at the same time, but with different configurations. like, you can build debug and release at the same time. if you would keep global state about which one you're building, it would break.